Fix Shortcut to toggle grid.

Carreau opened the pull request:
grid_size have been removed as part of #2144,

I'm not too sure how it was working, as shape seem to always be (-1, -1)
on the few examples I've tried; so replicate what the grid button does
seem to do

## Type of change

- [x] Bug-fix (non-breaking change which fixes an issue)

# How has this been tested?

$ napari

Check that ctrl-G change the icon on the grid button.  (I'm not too sure what this is supposed to do...)

I'm not too sure how to test that; but with the work on the action manager we should have more confidence that the buttons and shortcut do actually trigger the same effects.

Carreau committed:
Fix Shortcut to toggle grid.

grid_size have been removed as part of #2144,

I'm not too sure how it was working, as shape seem to always be (-1, -1)
on the few examples I've tried; so replicate what the grid button does
seem to do

sofroniewn reviewed (approved):
Yes, this looks good, definitely an outdated API. I'm surprised this wasn't caught by any of our existing tests. Specifically it should have been caught here https://github.com/napari/napari/blob/5a2b3c31b5641cd27a44adde11a0ed377922f789/napari/_tests/test_viewer.py#L37-L44 can you see why it was missed?

Carreau committed:
Fix non-running tests.

Also make which fucntion is tested a fixture in order to know which one
if failing; and fix failing case.

Carreau committed:
check layers number of methods

sofroniewn commented on napari/_tests/test_viewer.py:
```diff
@@ -33,15 +41,12 @@ def test_viewer(make_napari_viewer):
-        if func.__name__ == 'play':
-            continue
-        func(viewer)
+    if func.__name__ == 'toggle_fullscreen' and not os.getenv("CI"):
```
Can you pull this out into its own test, `tes_viewer_bindings`, that doesn't have any of the `assert` stuff above, no need for that to get tested each time

sofroniewn commented on napari/_tests/test_viewer.py:
```diff
@@ -12,8 +12,16 @@
 
-def test_viewer(make_napari_viewer):
+
+def test_len_methods_viewer():
```
Can you add a comment doc string about what the `18` is? (I guess number of keybindings on the viewer).

sofroniewn commented on napari/components/grid.py:
```diff
@@ -54,6 +54,8 @@ def actual_shape(self, nlayers: int = 1) -> Tuple[int, int]:
             Number of rows and columns in the grid.
         """
         if self.enabled:
+            if nlayers == 0:
```
fine with me

sofroniewn reviewed (commented):
> viewer.class_keymap is en empty dict, the bindings are on viewer model, not viewer, so the for loop does nothing.

Ah ok, clearly a bad test! nice catch. pulling this out into its own parametrized fixture makes sense, I think that could actually be its own test, separate from `test_viewer`, see comment above.

> the Image type layers has no functions; is that expected ? I don't see image specific buttons in the GUI; so that might be fine, but I want toget confirmation first;

That is right. I don't love `assert len(methods) in (14, 8, 17)` though, it feels a little magical and hard to remember where those numbers came from. Do you want to create a dict with the layer type and the number of expected methods and then check that it matches?

Carreau committed:
properly split test and number of methods

Carreau committed:
Parametrize on which method we test
